swing-lib/src/main/java/org/triplea/swing/JFrameBuilder.java

```diff
@@ -80,7 +98,26 @@ public JFrame build() {
     return frame;
   }
 
-  // TODO: unit-test this.
+  /**
+   * Returns the standard application icon typically displayed in a window's title bar.
+   */
+  public static Image getGameIcon(final Window frame) {
+    Image img = null;
+    try {
+      img = frame.getToolkit().getImage(JFrameBuilder.class.getResource("ta_icon.png"));
+    } catch (final Exception ex) {
+      log.log(Level.SEVERE, "ta_icon.png not loaded", ex);
+    }
+    final MediaTracker tracker = new MediaTracker(frame);
+    tracker.addImage(img, 0);
+    try {
+      tracker.waitForAll();
+    } catch (final InterruptedException ex) {
+      Thread.currentThread().interrupt();
+    }
+    return img;
```

Comment:
This whole method could be simplified to `return ImageIO.read(JFrameBuilder.class.getResource("ta_icon.png"));` in a try-catch statement.
Returning null when something went wrong.
See https://stackoverflow.com/a/21977133

Reply:
This code is only moved, further refactoring is not in scope

Reply:
That's fine, just wanted to point it out.
I think it's questionable to read images from memory on the Event Thread anyways.

Reply:
Follow-up is in: https://github.com/triplea-game/triplea/pull/4659